### DISCUSSION TITLE
Visualize attention maps on input spectrograms

### DISCUSSION THREAD

**Discussion by mmerler** [openai/whisper#2177] [Q&A]
Does anyone know how to visualize the encoder attention maps with respect to the input spectrograms? 
I'm interested in understanding which portions of the spectrogram a whisper-base fine-tuned model is focusing on when making a prediction.
I can extract the attention maps in the forward pass, each is 1500x1500, but I don't know how to map them back to the input spectrogram. 

Any ideas?

**Comment by mmerler** [comment]
basically the equivalent of Grad-Cam for audio with whisper?

**Comment by Coder1010ayush** [comment]
Any updates on this?
